lib/common/messenger.js

```diff
@@ -86,9 +86,16 @@ function messengerFactory(
 
         assert.ok(uri, 'uri');
 
-        var options = {
-            url: uri
-        };
+        var options;
+
+        if (typeof uri === 'string') {
+            options = {
+                url: uri
+            };
+        } else {
```

Comment:
Looks good. Just wondering if it would be a good idea to add add ```else if (Array.isArray(url))``` to explicitly declare that an Array is expected here, for code readability consideration. Looks good and :+1:  anyway.